src/main/java/gov/nasa/jpl/mbee/mdk/systems_reasoner/actions/ImportCSVAction.java

```diff
@@ -0,0 +1,416 @@
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private String handleSubProperty(List<Element> sortedColumns, String propertyName, Classifier parent) {
+        String[] subprops = propertyName.split("\\.");
+        propertyName = propertyName.replace(subprops[0] + ".", "");
+        String newTypeName = subprops[0];
+        Element property = getPropertyFromColumnName(newTypeName, parent);
+        if (property instanceof TypedElement) {
+            Type type = ((TypedElement) property).getType();
+            if (type instanceof Classifier) {
+                if (propertyName.contains(".")) {
+                    handleSubProperty(sortedColumns, propertyName, (Classifier) type);
+                } else {
+                    sortedColumns.add(getPropertyFromColumnName(propertyName, (Classifier) type));
+                }
+            }
+        }
+        return propertyName;
+    }
+
+    private Classifier createNewSubElement(String[] line, int index, Classifier generalClassifier) {
+
+        Classifier createdClassifier = (Classifier) CopyPasting.copyPasteElement(generalClassifier, container);
+        createdClassifier.getOwnedMember().clear();
+        createdClassifier.getGeneralization().clear();
+        Utils.createGeneralization(generalClassifier, createdClassifier);
+        createdClassifier.setName(line[index]);
+        Application.getInstance().getGUILog().log("Creating new " + generalClassifier.getName() + " named " + createdClassifier.getName() + " for line " + row + ".");
+        return createdClassifier;
+    }
+
+    private void setPropertyValue(String valueFromCSV, RedefinableElement el, HashMap<Property, Classifier> thisLinesClasses) {
+        if (el instanceof TypedElement) {
+            if (el.getOwner() instanceof Classifier) {
+                Classifier owner = findMatchingSubclass((Classifier) el.getOwner(), thisLinesClasses.values());
+                if (owner != null) {
+                    Property prop = UMLFactory.eINSTANCE.createProperty();
+                    prop.setType(((TypedElement) el).getType());
+                    prop.getRedefinedElement().add(el);
+                    prop.setOwner(owner);
+                    prop.setName(el.getName());
+                    ValueSpecification vs = null;
+                    Classifier linkedElement = null;
+                    if (((TypedElement) el).getType() instanceof Classifier) {
+                        linkedElement = findMatchingSubclass((Classifier) ((TypedElement) el).getType(), thisLinesClasses.values());
+                    }
+                    if (linkedElement == null) {
+                        if (((Property) el).getType() instanceof DataType) {
+                            if (!valueFromCSV.isEmpty()) {
+                                try {
+                                    if (((Property) el).getType().getID().equals("_16_5_1_12c903cb_1245415335546_479030_4092")) {
```

Comment:
IDs are constants and should be defined statically.

Example: public static final SOME_PROPERTY_TYPE_ID = "_16_5_1_12c903cb_1245415335546_479030_4092";